##### **Internship 2025 - Ouiam EL HAJJI**

## $$ \textit{Predictive and Exploratory Models in CNS tumors using ctDNA}  $$
<div align="center">

<h5>Department of Computational Biology and Medical Sciences  $$ \textit{University of Tokyo} $$ </h5>

</div>
________________________________________________________________________________________________________________________________________________

$$ \textit{Objective 1 : Predict Overall Survival} $$
**Goal** : Predict overall survival (OS_MONTHS, OS_STATUS) using genetic and clinical features.

**Pipeline** :
- Create patient × features matrix (mutation presence, TMB_NONSYNONYMOUS, SEX, CANCER_TYPE, SAMPLE_CLASS)
- Encode categorical variables
- Train/test split
- Apply models:
  • Classification: Logistic Regression, Random Forest, XGBoost, SVM
  • Regression: Linear Regression, GradientBoostingRegressor
  • Survival: Cox Model, Random Survival Forest
Explore:
- Key predictors of survival
- Linear vs non-linear models
- Which mutations drive short/long survival

### $$ \text{I.    Datasets managment (merging, cleaning,...)} $$

In [374]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

### **Patient dataset (survival, duration, sex)**

In [305]:
file_name = 'CerebrospinalFluidCirculatingTumor/data_clinical_patient.txt'

pd.set_option('display.max_columns', None)

df_cerebros_patient = pd.read_csv(file_name, sep = "\t", skiprows=4)
df_cerebros_patient

PATIENT_ID   OS_MONTHS   OS_STATUS     SEX
0    P-0042717    4.241049  1:DECEASED    Male
1    P-0053191   38.399579  1:DECEASED  Female
2    P-0074598    6.509518    0:LIVING    Male
3    P-0037509   11.243712  1:DECEASED    Male
4    P-0046067   58.618536    0:LIVING    Male
..         ...         ...         ...     ...
706  P-0006299  103.593385    0:LIVING  Female
707  P-0052502   53.687083    0:LIVING    Male
708  P-0075798   29.884604    0:LIVING    Male
709  P-0043477   10.980702  1:DECEASED  Female
710  P-0082845         NaN    0:LIVING    Male

[711 rows x 4 columns]

### **Patient sample dataset (TMB_NONSYNONYMOUS, CANCER_TYPE,...)**

In [142]:
file_name = 'CerebrospinalFluidCirculatingTumor/data_clinical_sample.txt'

pd.set_option('display.max_columns', None)

df_cerebros_sample = pd.read_csv(file_name, sep = "\t", skiprows = 4)
df_cerebros_sample

SAMPLE_ID PATIENT_ID ONCOTREE_CODE CTDNA_STATUS SOMATIC_STATUS  \
0     P-0042717-T03-IM6  P-0042717          LUNG     Positive        Matched   
1     P-0053191-T09-IM7  P-0053191          MNET     Positive        Matched   
2     P-0074598-T01-IM7  P-0074598          GNOS     Positive        Matched   
3     P-0037509-T02-IM6  P-0037509          GNOS     Positive        Matched   
4     P-0046067-T05-IM6  P-0046067         PRNET     Positive        Matched   
...                 ...        ...           ...          ...            ...   
1002  P-0001281-T03-IM6  P-0001281          GNOS     Negative        Matched   
1003  P-0049043-T01-IM6  P-0049043          EMBT     Positive        Matched   
1004  P-0071305-T02-IM7  P-0071305          EMBT     Negative        Matched   
1005  P-0082845-T01-IM7  P-0082845          EMBT     Negative        Matched   
1006  P-0043866-T03-IM6  P-0043866            TT     Positive        Matched   

      TMB_NONSYNONYMOUS                          CANCER_TYPE  \
0              0.100000                          Lung Cancer   
1              0.100000  Miscellaneous Neuroepithelial Tumor   
2              0.000000            Miscellaneous Brain Tumor   
3              0.200000            Cancer of Unknown Primary   
4              0.000000            Miscellaneous Brain Tumor   
...                 ...                                  ...   
1002           0.000000                               Glioma   
1003           0.033333                      Embryonal Tumor   
1004           0.000000                      Embryonal Tumor   
1005           0.000000                      Embryonal Tumor   
1006           0.000000                      Germ Cell Tumor   

                     CANCER_TYPE_DETAILED SAMPLE_CLASS  
0                                    Lung        ctDNA  
1     Miscellaneous Neuroepithelial Tumor        ctDNA  
2                             Glioma, NOS        ctDNA  
3                             Glioma, NOS        ctDNA  
4           Primary Neuroepithelial Tumor        ctDNA  
...                                   ...          ...  
1002                          Glioma, NOS        ctDNA  
1003                      Embryonal Tumor        ctDNA  
1004                      Embryonal Tumor        ctDNA  
1005                      Embryonal Tumor        ctDNA  
1006                             Teratoma        ctDNA  

[1007 rows x 9 columns]

### **Get unique values**

In [83]:
unique_values = df_cerebros['P-0054168-T01-IM6'].unique()
print(unique_values , len(df_cerebros.columns.unique()))

### **Mutation dataset : Mutation*Patient matrix**

In [347]:
file_name = 'CerebrospinalFluidCirculatingTumor/data_cna.txt'

pd.set_option('display.max_columns', None)

df_cerebros = pd.read_csv(file_name, sep = "\t", index_col = 'Hugo_Symbol')



df_cerebros

P-0071527-T01-IM7  P-0054168-T01-IM6  P-0064311-T02-IM7  \
Hugo_Symbol                                                            
TAP1                       0.0                0.0                0.0   
ERRFI1                     0.0                0.0                0.0   
STK19                      0.0                0.0                0.0   
CRKL                       0.0                0.0                0.0   
SCG5                       0.0                0.0                0.0   
...                        ...                ...                ...   
PDPK1                      0.0                0.0                0.0   
BAP1                       0.0                0.0                0.0   
SDHB                       0.0                0.0                0.0   
SDHD                       0.0                0.0                0.0   
PRKAR1A                    0.0                0.0                0.0   

             P-0080115-T03-IM7  P-0066284-T01-IM7  P-0035665-T07-IM6  \
Hugo_Symbol                                                            
TAP1                       0.0                0.0                0.0   
ERRFI1                     0.0                0.0                0.0   
STK19                      0.0                0.0                0.0   
CRKL                       0.0                0.0                0.0   
SCG5                       0.0                0.0                0.0   
...                        ...                ...                ...   
PDPK1                      0.0                0.0                0.0   
BAP1                       0.0                0.0                0.0   
SDHB                       0.0                0.0                0.0   
SDHD                       0.0                0.0                0.0   
PRKAR1A                    0.0                0.0                0.0   

             P-0057571-T03-IM7  P-0049043-T05-IM7  P-0064769-T05-IM7  \
Hugo_Symbol                                                            
TAP1                       0.0                0.0                0.0   
ERRFI1                     0.0                0.0                0.0   
STK19                      0.0                0.0                0.0   
CRKL                       0.0                0.0                0.0   
SCG5                       0.0                0.0                0.0   
...                        ...                ...                ...   
PDPK1                      0.0                0.0                0.0   
BAP1                       0.0                0.0                0.0   
SDHB                       0.0                0.0                0.0   
SDHD                       0.0                0.0                0.0   
PRKAR1A                    0.0                0.0                0.0   

             P-0005114-T09-IM7  P-0082740-T02-IM7  P-0038911-T03-IM7  \
Hugo_Symbol                                                            
TAP1                       0.0                0.0                0.0   
ERRFI1                     0.0                0.0                0.0   
STK19                      0.0                0.0                0.0   
CRKL                       0.0                0.0                0.0   
SCG5                       0.0                0.0                0.0   
...                        ...                ...                ...   
PDPK1                      0.0                0.0                0.0   
BAP1                       0.0                0.0                0.0   
SDHB                       0.0                0.0                0.0   
SDHD                       0.0                0.0                0.0   
PRKAR1A                    0.0                0.0                0.0   

             P-0064701-T02-IM7  P-0083253-T01-IM7  P-0065133-T01-IM7  \
Hugo_Symbol                                                            
TAP1                       0.0                0.0                0.0   
ERRFI1                     0.0                0.0

### **Transposing so i have the matrix in [ patient * mutation ] mode**

In [348]:
df_tr = df_cerebros.transpose()
df_tr
df_tr.reset_index
df_tr

Hugo_Symbol        TAP1  ERRFI1  STK19  CRKL  SCG5  STK11  MEN1  B2M  TAP2  \
P-0071527-T01-IM7   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
P-0054168-T01-IM6   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
P-0064311-T02-IM7   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
P-0080115-T03-IM7   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
P-0066284-T01-IM7   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
...                 ...     ...    ...   ...   ...    ...   ...  ...   ...   
P-0074634-T02-IM7   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
P-0059526-T04-IM7   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
P-0077891-T03-IM7   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
P-0055378-T07-IM7   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
P-0052694-T01-IM6   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   

Hugo_Symbol        PMAIP1  DUSP22  H3-3A  H3-3B  SMG1  CDC73  PIK3CA  PIK3CB  \
P-0071527-T01-IM7     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   
P-0054168-T01-IM6     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   
P-0064311-T02-IM7     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   
P-0080115-T03-IM7     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   
P-0066284-T01-IM7     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   
...                   ...     ...    ...    ...   ...    ...     ...     ...   
P-0074634-T02-IM7     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   
P-0059526-T04-IM7     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   
P-0077891-T03-IM7     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   
P-0055378-T07-IM7     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   
P-0052694-T01-IM6     0.0     NaN    0.0    0.0   NaN    0.0     0.0     0.0   

Hugo_Symbol        PIK3CD  MIB1  PIK3CG  CRBN  CBLB  IGF1R  FBXO11  NUP98  \
P-0071527-T01-IM7     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   
P-0054168-T01-IM6     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   
P-0064311-T02-IM7     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   
P-0080115-T03-IM7     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   
P-0066284-T01-IM7     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   
...                   ...   ...     ...   ...   ...    ...     ...    ...   
P-0074634-T02-IM7     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   
P-0059526-T04-IM7     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   
P-0077891-T03-IM7     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   
P-0055378-T07-IM7     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   
P-0052694-T01-IM6     0.0   NaN     0.0   NaN   NaN    0.0     NaN    NaN   

Hugo_Symbol        NUP93  SERP2  PIK3C2G  SMARCD1  PIK3C3  GATA2  GATA3  \
P-0071527-T01-IM7    0.0    NaN      0.0      0.0     0.0    0.0    0.0   
P-0054168-T01-IM6    0.0    NaN      0.0      0.0     0.0    0.0    0.0   
P-0064311-T02-IM7    0.0    NaN      0.0      0.0     0.0    0.0    0.0   
P-0080115-T03-IM7    0.0    NaN      0.0      0.0     0.0    0.0    0.0   
P-0066284-T01-IM7    0.0    NaN      0.0      0.0     0.0    0.0    0.0   
...                  ...    ...      ...      ...     ...    ...    ...   
P-0074634-T02-IM7    0.0    NaN      0.0      0.0     0.0    0.0    0.0   
P-0059526-T04-IM7    0.0    NaN      0.0      0.0     0.0    0.0    0.0   
P-0077891-T03-IM7    0.0    NaN      0.0      0.0     0.0    0.0    0.0   
P-0055378-T07-IM7    0.0    NaN      0.0      0.0     0.0    0.0    0.0   
P-0052694-T01-IM6    0.0    NaN      0.0      0.0     0.0    0.0    0.0   

Hugo_Symbol        GATA1  CCND1  MAX  CCND3  CCND2  PTEN  NTHL1  CTCF  IL7R  \
P-0071527-T01-IM7    0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0   
P-0054168-T01-IM6    2.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0   
P-0064311-T02-IM7    0.0    0.0  0.0   

In [263]:
unique_values = df_tr['SMG1'].unique()
print(unique_values , len(df_cerebros.columns.unique()))

[nan] 603


In [268]:
df_tr.columns

Index(['TAP1', 'ERRFI1', 'STK19', 'CRKL', 'SCG5', 'STK11', 'MEN1', 'B2M',
       'TAP2', 'PMAIP1',
       ...
       'PTPRD', 'GNA12', 'BRAF', 'FAM175A', 'SDHA', 'PDPK1', 'BAP1', 'SDHB',
       'SDHD', 'PRKAR1A'],
      dtype='object', name='Hugo_Symbol', length=701)

In [271]:
L= df_tr.columns
def get_MOST_NaN_columns(L):
    unique = 0
    l = 0
    LL = []
    for elt in L:
        count = df_tr[elt].isnull().sum()
        
        if count == 603 : 
            LL.append(elt)

    return LL, len(LL)
            
        
print(get_MOST_NaN_columns(L))

(['DUSP22', 'SMG1', 'MIB1', 'CRBN', 'CBLB', 'FBXO11', 'NUP98', 'SERP2', 'PDS5B', 'SETBP1', 'TYK2', 'BACH2', 'FAF1', 'KSR2', 'PIGA', 'ZEB2', 'ANKRD26', 'HIST1H2BG', 'MYBBP1A', 'ATXN2', 'ATP6AP1', 'HIST1H2BO', 'CIITA', 'SMC3', 'H1-3', 'JARID2', 'LTB', 'ZNF292', 'FOXO3', 'ARID5A', 'METTL23', 'RUNX1T1', 'KLF2', 'ETNK1', 'ATP6V1B2', 'P2RY8', 'YLPM1', 'SETD4', 'SETD5', 'SETD6', 'FANCL', 'H2AC17', 'PLCG1', 'DDX41', 'NCSTN', 'MGAM', 'ITPKB', 'IKZF3', 'IRF8', 'HDAC1', 'CD70', 'HDAC4', 'HDAC7', 'ZFHX4', 'FANCD2', 'MECOM', 'FANCG', 'FANCF', 'FANCE', 'FANCM', 'FANCI', 'SP140', 'TET3', 'IRF1', 'SRP72', 'MYO18A', 'SMARCA1', 'SBDS', 'PRKCB', 'CHD2', 'ACTG1', 'HIST1H1D', 'FURIN', 'MAP3K6', 'MAP3K7', 'H1-4', 'SETDB2', 'H1-B', 'SETD7', 'HIF1A', 'U2AF2', 'EGR2', 'EGR1', 'NFE2', 'H2AC6', 'CUX1', 'PTPN2', 'PTPN1', 'SRCAP', 'SETD1B', 'LCK', 'MKI67', 'MOB3B', 'BCORL1', 'BCR', 'STAT6', 'CD58', 'RPS26', 'EXOSC6', 'TBL1XR1', 'SGK1', 'DTX1', 'FAS', 'TERC', 'NT5C2', 'ARID3A', 'KLHL6', 'ARID3C', 'ARID3B', 'SAMHD1'

In [284]:
mutation_matrix = df_tr.drop(['DUSP22', 'SMG1', 'MIB1', 'CRBN', 'CBLB', 'FBXO11', 'NUP98', 'SERP2', 'PDS5B', 'SETBP1', 'TYK2', 'BACH2', 'FAF1', 'KSR2', 'PIGA', 'ZEB2', 'ANKRD26', 'HIST1H2BG', 'MYBBP1A', 'ATXN2', 'ATP6AP1', 'HIST1H2BO', 'CIITA', 'SMC3', 'H1-3', 'JARID2', 'LTB', 'ZNF292', 'FOXO3', 'ARID5A', 'METTL23', 'RUNX1T1', 'KLF2', 'ETNK1', 'ATP6V1B2', 'P2RY8', 'YLPM1', 'SETD4', 'SETD5', 'SETD6', 'FANCL', 'H2AC17', 'PLCG1', 'DDX41', 'NCSTN', 'MGAM', 'ITPKB', 'IKZF3', 'IRF8', 'HDAC1', 'CD70', 'HDAC4', 'HDAC7', 'ZFHX4', 'FANCD2', 'MECOM', 'FANCG', 'FANCF', 'FANCE', 'FANCM', 'FANCI', 'SP140', 'TET3', 'IRF1', 'SRP72', 'MYO18A', 'SMARCA1', 'SBDS', 'PRKCB', 'CHD2', 'ACTG1', 'HIST1H1D', 'FURIN', 'MAP3K6', 'MAP3K7', 'H1-4', 'SETDB2', 'H1-B', 'SETD7', 'HIF1A', 'U2AF2', 'EGR2', 'EGR1', 'NFE2', 'H2AC6', 'CUX1', 'PTPN2', 'PTPN1', 'SRCAP', 'SETD1B', 'LCK', 'MKI67', 'MOB3B', 'BCORL1', 'BCR', 'STAT6', 'CD58', 'RPS26', 'EXOSC6', 'TBL1XR1', 'SGK1', 'DTX1', 'FAS', 'TERC', 'NT5C2', 'ARID3A', 'KLHL6', 'ARID3C', 'ARID3B', 'SAMHD1', 'ARHGEF28', 'SETD1A', 'UBR5', 'HIST1H2AL', 'MPEG1', 'CD28', 'DDX3X', 'SOCS3', 'SOCS2', 'ROBO1', 'SETD3', 'TRAF3', 'EP400', 'ARID4B', 'PCBP1', 'SMC1A', 'NXF1', 'STAG1', 'RHOH', 'HIST1H2BC', 'HIST1H2BJ', 'HIST1H2BK', 'ETV5', 'ETV4', 'ZMYM3', 'VAV1', 'VAV2', 'H2AC16', 'NCOR2', 'BTG1', 'PHIP', 'H2B17', 'HIST1H2AG', 'H2BC4', 'H2BC8', 'HIST1H1E', 'HIST1H1B', 'RPS15', 'ZNF318', 'RPS19', 'JUNB', 'BCL11B', 'HIST1H2AM', 'ARID4A', 'HIST1H2AC', 'XBP1', 'TNFRSF1B', 'ESCO2', 'LRP1B', 'GNA13', 'TRAF5', 'GNA12'], axis=1)

mutation_matrix.index

Index(['P-0071527-T01-IM7', 'P-0054168-T01-IM6', 'P-0064311-T02-IM7',
       'P-0080115-T03-IM7', 'P-0066284-T01-IM7', 'P-0035665-T07-IM6',
       'P-0057571-T03-IM7', 'P-0049043-T05-IM7', 'P-0064769-T05-IM7',
       'P-0005114-T09-IM7',
       ...
       'P-0058474-T04-IM7', 'P-0052145-T05-IM7', 'P-0046319-T03-IM7',
       'P-0035665-T13-IM7', 'P-0082766-T01-IM7', 'P-0074634-T02-IM7',
       'P-0059526-T04-IM7', 'P-0077891-T03-IM7', 'P-0055378-T07-IM7',
       'P-0052694-T01-IM6'],
      dtype='object', length=603)

#### Creating copy of **mutation_matrix** with `Patient_ID` as an index instead of `Tumor_ID`

In [356]:
mutation_matrix_copy = mutation_matrix.copy()

#mutation_matrix_copy
def Keep_patient_ID(df):
    df.index = df.index.map(lambda x: x[:9])
    return df
Keep_patient_ID(mutation_matrix_copy)

Hugo_Symbol  TAP1  ERRFI1  STK19  CRKL  SCG5  STK11  MEN1  B2M  TAP2  PMAIP1  \
P-0071527     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   
P-0054168     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   
P-0064311     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   
P-0080115     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   
P-0066284     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   
...           ...     ...    ...   ...   ...    ...   ...  ...   ...     ...   
P-0074634     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   
P-0059526     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   
P-0077891     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   
P-0055378     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   
P-0052694     0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0     0.0   

Hugo_Symbol  H3-3A  H3-3B  CDC73  PIK3CA  PIK3CB  PIK3CD  PIK3CG  IGF1R  \
P-0071527      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
P-0054168      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
P-0064311      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
P-0080115      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
P-0066284      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
...            ...    ...    ...     ...     ...     ...     ...    ...   
P-0074634      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
P-0059526      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
P-0077891      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
P-0055378      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
P-0052694      0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   

Hugo_Symbol  NUP93  PIK3C2G  SMARCD1  PIK3C3  GATA2  GATA3  GATA1  CCND1  MAX  \
P-0071527      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0   
P-0054168      0.0      0.0      0.0     0.0    0.0    0.0    2.0    0.0  0.0   
P-0064311      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0   
P-0080115      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0   
P-0066284      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0   
...            ...      ...      ...     ...    ...    ...    ...    ...  ...   
P-0074634      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0   
P-0059526      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0   
P-0077891      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0   
P-0055378      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0   
P-0052694      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0   

Hugo_Symbol  CCND3  CCND2  PTEN  NTHL1  CTCF  IL7R  CDKN1B  CDKN1A  RIT1  \
P-0071527      0.0    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   
P-0054168      0.0    0.0   0.0    0.0   0.0   0.0    -2.0     0.0   0.0   
P-0064311      0.0    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   
P-0080115      0.0    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   
P-0066284      0.0    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   
...            ...    ...   ...    ...   ...   ...     ...     ...   ...   
P-0074634      0.0    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   
P-0059526      0.0    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   
P-0077891      0.0    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   
P-0055378      0.0    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   
P-0052694      0.0    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   

Hugo_Symbol  MLL3  MLL2  MLL4  XPO1  CSDE1  ETAA1  ARID2  MST1  SOX17  JAK2  \
P-0071527     0.0   0.0   0.0   0.0    0.0    0.0    0.0   0.0    0.0   0.0   
P-0054168     0.0   0.0   0.0   0.0    0.0    0.0    0.0   0.0    0.0   2.0   
P-0064311     0

### **Data gene panel matrix**

In [56]:
file_name = 'CerebrospinalFluidCirculatingTumor/data_gene_panel_matrix.txt'

pd.set_option('display.max_columns', None)

df_cerebros = pd.read_csv(file_name, sep = "\t")

df_cerebros




SAMPLE_ID  mutations     gistic structural_variants
0    P-0042717-T03-IM6  IMPACT468  IMPACT468           IMPACT468
1    P-0053191-T09-IM7  IMPACT505  IMPACT505           IMPACT505
2    P-0074598-T01-IM7  IMPACT505  IMPACT505           IMPACT505
3    P-0037509-T02-IM6  IMPACT468  IMPACT468           IMPACT468
4    P-0046067-T05-IM6  IMPACT468  IMPACT468           IMPACT468
..                 ...        ...        ...                 ...
598  P-0071933-T02-IM7  IMPACT505  IMPACT505           IMPACT505
599  P-0058734-T06-IM7  IMPACT505  IMPACT505           IMPACT505
600  P-0058734-T08-IM7  IMPACT505  IMPACT505           IMPACT505
601  P-0049043-T01-IM6  IMPACT468  IMPACT468           IMPACT468
602  P-0043866-T03-IM6  IMPACT468  IMPACT468           IMPACT468

[603 rows x 4 columns]

### **Mutation + features dataset (without empty columns)**

In [158]:
df_cleaned = df_cerebros_mut.dropna(axis=1, how='all')
df_cleaned

Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \
0           TP53            7157  MSKCC     GRCh37         17         7577121   
1           IDH1            3417  MSKCC     GRCh37          2       209113112   
2           PTEN            5728  MSKCC     GRCh37         10        89717609   
3           ATRX             546  MSKCC     GRCh37          X        76939783   
4          PRDM1             639  MSKCC     GRCh37          6       106554349   
...          ...             ...    ...        ...        ...             ...   
3939        TP53            7157  MSKCC     GRCh37         17         7577556   
3940      NOTCH1            4851  MSKCC     GRCh37          9       139413070   
3941        ATRX             546  MSKCC     GRCh37          X        76938752   
3942      PIK3R1            5295  MSKCC     GRCh37          5        67576365   
3943      PIK3R1            5295  MSKCC     GRCh37          5        67589621   

      End_Position Strand              Consequence Variant_Classification  \
0          7577121      +         missense_variant      Missense_Mutation   
1        209113112      +         missense_variant      Missense_Mutation   
2         89717609      +  splice_acceptor_variant            Splice_Site   
3         76939783      +              stop_gained      Nonsense_Mutation   
4        106554349      +         missense_variant      Missense_Mutation   
...            ...    ...                      ...                    ...   
3939       7577556      +         missense_variant      Missense_Mutation   
3940     139413072      +         inframe_deletion           In_Frame_Del   
3941      76938752      +              stop_gained      Nonsense_Mutation   
3942      67576365      +         missense_variant      Missense_Mutation   
3943      67589621      +              stop_gained      Nonsense_Mutation   

     Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2  \
0             SNP                G                 G                 A   
1             SNP                C                 C                 T   
2             SNP                G                 G                 C   
3             SNP                G                 G                 C   
4             SNP                C                 C                 T   
...           ...              ...               ...               ...   
3939          SNP                C                 C                 A   
3940          DEL              AGA               AGA                 -   
3941          SNP                G                 G                 A   
3942          SNP                G                 G                 A   
3943          SNP                G                 G                 T   

         dbSNP_RS Tumor_Sample_Barcode Validation_Status Mutation_Status  \
0     rs121913343    P-0086233-T01-IM7           Unknown         SOMATIC   
1     rs121913500    P-0086233-T01-IM7           Unknown         SOMATIC   
2     rs876661024    P-0086233-T01-IM7           Unknown         SOMATIC   
3             NaN    P-0086233-T01-IM7           Unknown         SOMATIC   
4             NaN    P-0086233-T01-IM7           Unknown         SOMATIC   
...           ...                  ...               ...             ...   
3939  rs121912655    P-0026227-T02-IM6           Unknown         SOMATIC   
3940          NaN    P-0026227-T02-IM6           Unknown         SOMATIC   
3941          NaN    P-0026227-T02-IM6           Unknown         SOMATIC   
3942          NaN    P-0026227-T02-IM6           Unknown         SOMATIC   
3943          NaN    P-0026227-T02-IM6           Unknown         SOMATIC   

           Score  t_ref_count  t_alt_count  n_ref_count  n_alt_count  \
0     MSK-IMPACT           13           39          430            0   
1     MSK-IMPACT           45           53          400            0   
2     MSK-IMPACT           61           55          339            0   
3     MSK-IMPACT 

#### $${I. a. Merging} $$

#### **Merging `Patient info` (PATIENT_ID, OS_MONTHS, OS_STATUS, SEX) and `Sample info` (TMB, Cancer_type,...)**

In [238]:
# S'assurer que PATIENT_ID est bien une chaîne de caractères dans les deux DataFrames
df_cerebros_sample["PATIENT_ID"] = df_cerebros_sample["PATIENT_ID"].astype(str)
df_cerebros_patient["PATIENT_ID"] = df_cerebros_patient["PATIENT_ID"].astype(str)

# Sélectionner uniquement les colonnes utiles du patient
patient_info = df_cerebros_patient[["PATIENT_ID", "OS_MONTHS", "OS_STATUS", "SEX"]]

# Fusionner avec df_cerebros_sample
df_merged = df_cerebros_sample.merge(patient_info, on="PATIENT_ID", how="left")
df_merged = df_merged.rename(columns={'SAMPLE_ID': 'Tumor_Sample_Barcode'})

df_merged = df_merged.dropna()

In [92]:
print("Nombre de lignes initial :", len(df_merged))
print("Nombre de lignes sans NaN :", len(df_merged.dropna()))
print("Lignes supprimées :", len(df_merged) - len(df_merged.dropna()))

Nombre de lignes initial : 1007
Nombre de lignes sans NaN : 939
Lignes supprimées : 68


In [93]:
df_cleaned = df_merged.dropna()

#### **Matching mutations to `Patient` & `Sample` info**

In [332]:
df_cLeaned= df_cleaned.copy()
df_merged = df_merged.copy()
df_cleaned = df_cleaned.rename(columns={'PATIENT_ID': 'Tumor_Sample_Barcode'})

df_merged['Tumor_Sample_Barcode'] = df_merged['Tumor_Sample_Barcode'].astype(str).str.strip()
df_cleaned['Tumor_Sample_Barcode'] = df_cleaned['Tumor_Sample_Barcode'].astype(str).str.strip()

df_final = df_cleaned.merge(df_merged, on='Tumor_Sample_Barcode', how='left')

df_final

Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \
0           TP53            7157  MSKCC     GRCh37         17         7577121   
1           IDH1            3417  MSKCC     GRCh37          2       209113112   
2           PTEN            5728  MSKCC     GRCh37         10        89717609   
3           ATRX             546  MSKCC     GRCh37          X        76939783   
4          PRDM1             639  MSKCC     GRCh37          6       106554349   
...          ...             ...    ...        ...        ...             ...   
3939        TP53            7157  MSKCC     GRCh37         17         7577556   
3940      NOTCH1            4851  MSKCC     GRCh37          9       139413070   
3941        ATRX             546  MSKCC     GRCh37          X        76938752   
3942      PIK3R1            5295  MSKCC     GRCh37          5        67576365   
3943      PIK3R1            5295  MSKCC     GRCh37          5        67589621   

      End_Position Strand              Consequence Variant_Classification  \
0          7577121      +         missense_variant      Missense_Mutation   
1        209113112      +         missense_variant      Missense_Mutation   
2         89717609      +  splice_acceptor_variant            Splice_Site   
3         76939783      +              stop_gained      Nonsense_Mutation   
4        106554349      +         missense_variant      Missense_Mutation   
...            ...    ...                      ...                    ...   
3939       7577556      +         missense_variant      Missense_Mutation   
3940     139413072      +         inframe_deletion           In_Frame_Del   
3941      76938752      +              stop_gained      Nonsense_Mutation   
3942      67576365      +         missense_variant      Missense_Mutation   
3943      67589621      +              stop_gained      Nonsense_Mutation   

     Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2  \
0             SNP                G                 G                 A   
1             SNP                C                 C                 T   
2             SNP                G                 G                 C   
3             SNP                G                 G                 C   
4             SNP                C                 C                 T   
...           ...              ...               ...               ...   
3939          SNP                C                 C                 A   
3940          DEL              AGA               AGA                 -   
3941          SNP                G                 G                 A   
3942          SNP                G                 G                 A   
3943          SNP                G                 G                 T   

         dbSNP_RS Tumor_Sample_Barcode Validation_Status Mutation_Status  \
0     rs121913343    P-0086233-T01-IM7           Unknown         SOMATIC   
1     rs121913500    P-0086233-T01-IM7           Unknown         SOMATIC   
2     rs876661024    P-0086233-T01-IM7           Unknown         SOMATIC   
3             NaN    P-0086233-T01-IM7           Unknown         SOMATIC   
4             NaN    P-0086233-T01-IM7           Unknown         SOMATIC   
...           ...                  ...               ...             ...   
3939  rs121912655    P-0026227-T02-IM6           Unknown         SOMATIC   
3940          NaN    P-0026227-T02-IM6           Unknown         SOMATIC   
3941          NaN    P-0026227-T02-IM6           Unknown         SOMATIC   
3942          NaN    P-0026227-T02-IM6           Unknown         SOMATIC   
3943          NaN    P-0026227-T02-IM6           Unknown         SOMATIC   

           Score  t_ref_count  t_alt_count  n_ref_count  n_alt_count  \
0     MSK-IMPACT           13           39          430            0   
1     MSK-IMPACT           45           53          400            0   
2     MSK-IMPACT           61           55          339            0   
3     MSK-IMPACT 

In [205]:
unique = df_final['PATIENT_ID'].unique()
print(len(unique))


341


68.2

#### $${I. b. Pre-processing : analysis of columns} $$

In [173]:
L = df_final.columns
def get_useless_column(L):
    unique = 0
    l = 0
    LL = []
    for elt in L:
        unique = df_final[elt].unique()
        l = len(unique)
        if l == 2 : 
            LL.append(elt)

    return LL, len(LL)
            
        
print(get_useless_column(L))


(['Center', 'NCBI_Build', 'Strand', 'Validation_Status', 'Mutation_Status', 'Score', 'GENE_IN_ONCOKB', 'Annotation_Status', 'CTDNA_STATUS', 'SOMATIC_STATUS', 'SAMPLE_CLASS'], 11)


In [194]:
def get_MOST_NaN_columns(L):
    unique = 0
    l = 0
    LL = []
    for elt in L:
        count = df_final[elt].isnull().sum()
        
        if count <= 788 : 
            LL.append(elt)

    return LL, len(LL)
            
        
print(get_MOST_NaN_columns(L))

(['Hugo_Symbol', 'Entrez_Gene_Id', 'Center', 'NCBI_Build', 'Chromosome', 'Start_Position', 'End_Position', 'Strand', 'Consequence', 'Variant_Classification', 'Variant_Type', 'Reference_Allele', 'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'Tumor_Sample_Barcode', 'Validation_Status', 'Mutation_Status', 'Score', 't_ref_count', 't_alt_count', 'n_ref_count', 'n_alt_count', 'HGVSc', 'HGVSp', 'HGVSp_Short', 'Transcript_ID', 'RefSeq', 'Protein_position', 'Codons', 'Exon_Number', 'GENE_IN_ONCOKB', 'MUTATION_EFFECT', 'ONCOGENIC', 'VARIANT_IN_ONCOKB', 'VF', 'Annotation_Status', 'PATIENT_ID', 'ONCOTREE_CODE', 'CTDNA_STATUS', 'SOMATIC_STATUS', 'TMB_NONSYNONYMOUS', 'CANCER_TYPE', 'CANCER_TYPE_DETAILED', 'SAMPLE_CLASS', 'OS_MONTHS', 'OS_STATUS', 'SEX'], 47)


In [184]:
def get_complete_columns(L):
    unique = 0
    l = 0
    LL = []
    for elt in L:
        count = df_final[elt].isnull().sum()
        
        if count == 0 : 
            LL.append(elt)

    return LL, len(LL)
            
        
print(get_complete_columns(L))

(['Hugo_Symbol', 'Entrez_Gene_Id', 'Center', 'NCBI_Build', 'Chromosome', 'Start_Position', 'End_Position', 'Strand', 'Consequence', 'Variant_Classification', 'Variant_Type', 'Reference_Allele', 'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'Tumor_Sample_Barcode', 'Validation_Status', 'Mutation_Status', 'Score', 't_ref_count', 't_alt_count', 'n_ref_count', 'n_alt_count', 'Transcript_ID', 'GENE_IN_ONCOKB', 'MUTATION_EFFECT', 'ONCOGENIC', 'VARIANT_IN_ONCOKB', 'VF', 'Annotation_Status', 'PATIENT_ID', 'ONCOTREE_CODE', 'CTDNA_STATUS', 'SOMATIC_STATUS', 'TMB_NONSYNONYMOUS', 'CANCER_TYPE', 'CANCER_TYPE_DETAILED', 'SAMPLE_CLASS', 'OS_STATUS', 'SEX'], 39)


### $$ \text{II. Random Forest} $$
#### $$ \textit{ Random Forest on Mutation+Features dataset} $$

In [246]:
def get_20_percent_patient(df):
    counting_patient = 0
    for i in range(len(df['PATIENT_ID'])):
        if df.iloc[i-1]['PATIENT_ID']!=df.iloc[i]['PATIENT_ID']:
            
            counting_patient += 1
            if counting_patient == 120:
                return i

get_20_percent_patient(df_final)   
                       

709

In [248]:
import category_encoders as ce
encoder = ce.OrdinalEncoder()
X = df_final.drop('OS_STATUS', axis=1)
y = df_final['OS_STATUS']

X_encoded = encoder.fit_transform(X)

clf = RandomForestClassifier(random_state=42)
clf.fit(X_encoded, y)

from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# 1. Séparer train/test (si pas encore fait)
X_train = X_encoded.iloc[:607]
X_test = X_encoded.iloc[607:]

y_train = y.iloc[:607]
y_test = y.iloc[607:]

# 2. Réentraîner le modèle
clf = RandomForestClassifier(random_state=42)
clf.fit(X_train, y_train)

# 3. Prédictions
y_pred = clf.predict(X_test)

# 4. Évaluation
print("=== Accuracy ===")
print(f"Accuracy: {accuracy_score(y_test, y_pred):.4f}")

print("\n=== Rapport de classification ===")
print(classification_report(y_test, y_pred))

print("\n=== Matrice de confusion ===")
print(confusion_matrix(y_test, y_pred))


=== Accuracy ===
Accuracy: 0.8616

=== Rapport de classification ===
              precision    recall  f1-score   support

    0:LIVING       0.41      0.14      0.21       434
  1:DECEASED       0.88      0.97      0.92      2903

    accuracy                           0.86      3337
   macro avg       0.65      0.56      0.57      3337
weighted avg       0.82      0.86      0.83      3337


=== Matrice de confusion ===
[[  62  372]
 [  90 2813]]


#### $$ \textit{ Random Forest on only Features dataset} $$

In [379]:
X = df_merged.drop('OS_STATUS', axis=1)
y = df_merged['OS_STATUS']




# 3. Encoder
X_encoded = encoder.fit_transform(X)
X_train, X_test, y_train, y_test = train_test_split(X_encoded, y, test_size=0.2, random_state=42)

clf = RandomForestClassifier(n_estimators=300, random_state=42)
clf.fit(X_train, y_train)
y_pred = clf.predict(X_test)




# 4. Évaluation
print("=== Accuracy ===")
print(f"Accuracy: {accuracy_score(y_test, y_pred):.4f}")

print("\n=== Rapport de classification ===")
print(classification_report(y_test, y_pred))

print("\n=== Matrice de confusion ===")
print(confusion_matrix(y_test, y_pred))


=== Accuracy ===
Accuracy: 0.8245

=== Rapport de classification ===
              precision    recall  f1-score   support

    0:LIVING       0.76      0.73      0.74        66
  1:DECEASED       0.86      0.88      0.87       122

    accuracy                           0.82       188
   macro avg       0.81      0.80      0.81       188
weighted avg       0.82      0.82      0.82       188


=== Matrice de confusion ===
[[ 48  18]
 [ 15 107]]


#### $$ \textit{ Random Forest on Mutation profile dataset only} $$

In [370]:
#mutation_matrix_copy = mutation_matrix_copy.reset_index()
mutation_matrix_patient_ID = Keep_patient_ID(mutation_matrix_copy)
mutation_matrix_patient_ID.columns
mutation_matrix_patient_ID = mutation_matrix_patient_ID.reset_index()
mutation_matrix_patient_ID


Hugo_Symbol      index  TAP1  ERRFI1  STK19  CRKL  SCG5  STK11  MEN1  B2M  \
0            P-0071527   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   
1            P-0054168   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   
2            P-0064311   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   
3            P-0080115   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   
4            P-0066284   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   
..                 ...   ...     ...    ...   ...   ...    ...   ...  ...   
598          P-0074634   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   
599          P-0059526   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   
600          P-0077891   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   
601          P-0055378   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   
602          P-0052694   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   

Hugo_Symbol  TAP2  PMAIP1  H3-3A  H3-3B  CDC73  PIK3CA  PIK3CB  PIK3CD  \
0             0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   
1             0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   
2             0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   
3             0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   
4             0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   
..            ...     ...    ...    ...    ...     ...     ...     ...   
598           0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   
599           0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   
600           0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   
601           0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   
602           0.0     0.0    0.0    0.0    0.0     0.0     0.0     0.0   

Hugo_Symbol  PIK3CG  IGF1R  NUP93  PIK3C2G  SMARCD1  PIK3C3  GATA2  GATA3  \
0               0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   
1               0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   
2               0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   
3               0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   
4               0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   
..              ...    ...    ...      ...      ...     ...    ...    ...   
598             0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   
599             0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   
600             0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   
601             0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   
602             0.0    0.0    0.0      0.0      0.0     0.0    0.0    0.0   

Hugo_Symbol  GATA1  CCND1  MAX  CCND3  CCND2  PTEN  NTHL1  CTCF  IL7R  CDKN1B  \
0              0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0     0.0   
1              2.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0    -2.0   
2              0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0     0.0   
3              0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0     0.0   
4              0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0     0.0   
..             ...    ...  ...    ...    ...   ...    ...   ...   ...     ...   
598            0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0     0.0   
599            0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0     0.0   
600            0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0     0.0   
601            0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0     0.0   
602            0.0    0.0  0.0    0.0    0.0   0.0    0.0   0.0   0.0     0.0   

Hugo_Symbol  CDKN1A  RIT1  MLL3  MLL2  MLL4  XPO1  CSDE1  ETAA1  ARID2  MST1  \
0               0.0   0.0   0.0   0.0   0.0   0.0    0.0    0.0    0.0   0.0   
1               0.0   0.0   0.0   0.0   0.0   0.0    0.0    0.0    0.0   0.0   
2               0.0   0.0   0.0   0.0   0.0   0.

In [371]:

mutation_matrix_patient_ID = mutation_matrix_patient_ID.rename(columns={'index': 'PATIENT_ID'})
mutation_matrix_patient_ID.columns

Index(['PATIENT_ID', 'TAP1', 'ERRFI1', 'STK19', 'CRKL', 'SCG5', 'STK11',
       'MEN1', 'B2M', 'TAP2',
       ...
       'PTPRS', 'PTPRD', 'BRAF', 'FAM175A', 'SDHA', 'PDPK1', 'BAP1', 'SDHB',
       'SDHD', 'PRKAR1A'],
      dtype='object', name='Hugo_Symbol', length=540)

In [372]:
mutation_matrix_patient_ID = mutation_matrix_patient_ID.rename(columns={'index': 'PATIENT_ID'})

df_cerebros_patient['PATIENT_ID'] = df_cerebros_patient['PATIENT_ID'].astype(str).str.strip()
mutation_matrix_patient_ID['PATIENT_ID'] = mutation_matrix_patient_ID['PATIENT_ID'].astype(str).str.strip()

mutation_matrix_with_class = mutation_matrix_patient_ID.merge(df_cerebros_patient, on='PATIENT_ID', how='left')

mutation_matrix_with_class

PATIENT_ID  TAP1  ERRFI1  STK19  CRKL  SCG5  STK11  MEN1  B2M  TAP2  \
0    P-0071527   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
1    P-0054168   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
2    P-0064311   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
3    P-0080115   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
4    P-0066284   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
..         ...   ...     ...    ...   ...   ...    ...   ...  ...   ...   
598  P-0074634   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
599  P-0059526   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
600  P-0077891   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
601  P-0055378   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   
602  P-0052694   0.0     0.0    0.0   0.0   0.0    0.0   0.0  0.0   0.0   

     PMAIP1  H3-3A  H3-3B  CDC73  PIK3CA  PIK3CB  PIK3CD  PIK3CG  IGF1R  \
0       0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
1       0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
2       0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
3       0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
4       0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
..      ...    ...    ...    ...     ...     ...     ...     ...    ...   
598     0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
599     0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
600     0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
601     0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   
602     0.0    0.0    0.0    0.0     0.0     0.0     0.0     0.0    0.0   

     NUP93  PIK3C2G  SMARCD1  PIK3C3  GATA2  GATA3  GATA1  CCND1  MAX  CCND3  \
0      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0    0.0   
1      0.0      0.0      0.0     0.0    0.0    0.0    2.0    0.0  0.0    0.0   
2      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0    0.0   
3      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0    0.0   
4      0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0    0.0   
..     ...      ...      ...     ...    ...    ...    ...    ...  ...    ...   
598    0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0    0.0   
599    0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0    0.0   
600    0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0    0.0   
601    0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0    0.0   
602    0.0      0.0      0.0     0.0    0.0    0.0    0.0    0.0  0.0    0.0   

     CCND2  PTEN  NTHL1  CTCF  IL7R  CDKN1B  CDKN1A  RIT1  MLL3  MLL2  MLL4  \
0      0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   0.0   0.0   0.0   
1      0.0   0.0    0.0   0.0   0.0    -2.0     0.0   0.0   0.0   0.0   0.0   
2      0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   0.0   0.0   0.0   
3      0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   0.0   0.0   0.0   
4      0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   0.0   0.0   0.0   
..     ...   ...    ...   ...   ...     ...     ...   ...   ...   ...   ...   
598    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   0.0   0.0   0.0   
599    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   0.0   0.0   0.0   
600    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   0.0   0.0   0.0   
601    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   0.0   0.0   0.0   
602    0.0   0.0    0.0   0.0   0.0     0.0     0.0   0.0   0.0   0.0   0.0   

     XPO1  CSDE1  ETAA1  ARID2  MST1  SOX17  JAK2  H3F3B  NF1  JAK1  GTF2I  \
0     0.0    0.0    0.0    0.0   0.0    0.0   0.0    0.0  0.0   0.0    0.0   
1     0.0    0.0    0.0    0.0   0.0    0.0   2.0    0.0  0.0   0.0    0.0   
2     0.0    0.0    0.0    0.0   0.0    0.0   0.0    0.0  

In [376]:
X = mutation_matrix_with_class.drop(['OS_STATUS','OS_MONTHS','SEX'], axis=1)
y = mutation_matrix_with_class['OS_STATUS']




# 3. Encoder
X_encoded = encoder.fit_transform(X)
X_train, X_test, y_train, y_test = train_test_split(X_encoded, y, test_size=0.2, random_state=42)

clf = RandomForestClassifier(n_estimators=300, random_state=42)
clf.fit(X_train, y_train)
y_pred = clf.predict(X_test)



# 4. Évaluation
print("=== Accuracy ===")
print(f"Accuracy: {accuracy_score(y_test, y_pred):.4f}")

print("\n=== Rapport de classification ===")
print(classification_report(y_test, y_pred))

print("\n=== Matrice de confusion ===")
print(confusion_matrix(y_test, y_pred))


=== Accuracy ===
Accuracy: 0.6364

=== Rapport de classification ===
              precision    recall  f1-score   support

    0:LIVING       0.40      0.49      0.44        35
  1:DECEASED       0.77      0.70      0.73        86

    accuracy                           0.64       121
   macro avg       0.58      0.59      0.58       121
weighted avg       0.66      0.64      0.65       121


=== Matrice de confusion ===
[[17 18]
 [26 60]]


In [378]:
X = mutation_matrix_with_class.drop(['OS_STATUS', 'OS_MONTHS', 'SEX'], axis=1)
y = mutation_matrix_with_class['OS_STATUS']




# 3. Encoder
X_encoded = encoder.fit_transform(X)
X_train, X_test, y_train, y_test = train_test_split(X_encoded, y, test_size=0.2, random_state=42)

clf = RandomForestClassifier(n_estimators=300, random_state=42)
clf.fit(X_train, y_train)
y_pred = clf.predict(X_test)


# 4. Évaluation
print("=== Accuracy ===")
print(f"Accuracy: {accuracy_score(y_test, y_pred):.4f}")

print("\n=== Rapport de classification ===")
print(classification_report(y_test, y_pred))

print("\n=== Matrice de confusion ===")
print(confusion_matrix(y_test, y_pred))


=== Accuracy ===
Accuracy: 0.6364

=== Rapport de classification ===
              precision    recall  f1-score   support

    0:LIVING       0.40      0.49      0.44        35
  1:DECEASED       0.77      0.70      0.73        86

    accuracy                           0.64       121
   macro avg       0.58      0.59      0.58       121
weighted avg       0.66      0.64      0.65       121


=== Matrice de confusion ===
[[17 18]
 [26 60]]


#### $$ \text{Conclusion so far :} $$

| dataset features         | RF Accuracy            |          0:LIVING Class precision        |
|----------------|-----------------------------------|---------------------------------|
| Mutation profile (mut*pat matrix)         | 63%          |       40%                      |
| Clinical features          | 82% |               76%              |
| Both (merged)     | 86%         |       41%                         |